Sources/NIO/ByteBuffer-aux.swift

```diff
@@ -373,6 +373,35 @@ extension ByteBuffer {
         self._moveWriterIndex(forwardBy: written)
         return written
     }
+    
+    /// Writes `byte` `count` times. Moves the writer index forward by the number of bytes written.
+    ///
+    /// - parameter byte: The `UInt8` byte to repeat.
+    /// - parameter count: How many times to repeat the given `byte`
+    /// - returns: How many bytes were written.
+    @discardableResult
+    public mutating func writeRepeatingByte(_ byte: UInt8, count: Int) -> Int {
+        let written = self.setRepeatingByte(byte, count: count, at: self.writerIndex)
+        self._moveWriterIndex(forwardBy: written)
+        return written
+    }
+    
+    /// Sets the given `byte` `count` times at the given `index`. Will reserve more memory if necessary. Does not move the writer index.
+    ///
+    /// - parameter byte: The `UInt8` byte to repeat.
+    /// - parameter count: How many times to repeat the given `byte`
+    /// - returns: How many bytes were written.
+    @discardableResult
+    public mutating func setRepeatingByte(_ byte: UInt8, count: Int, at index: Int) -> Int {
+        precondition(count >= 0, "Can't write fewer than 0 bytes")
+        self.reserveCapacity(index + count)
+        self.withVeryUnsafeMutableBytes { pointer in
+            precondition(pointer.count >= count)
```

Comment:
this `precondition` is in the wrong place. It needs to be right in front of `memset` asserting the correct pointer has enough bytes left.

Reply:
Probably something like

```swift
let dest = pointer[index...].base
preconditon(dest.count >= count)
memset(dest.baseAddress!, CInt(byte), size_t(count))
```

@Davidde94 also make sure to use the `CInt` (and not `Int32`) constructor as well as the `size_t` one. If you don't you're assuming the following things to be true:

- `CInt` == `Int32`
- `size_t` == `Int`

those are true for 64bit macOS but not on 32 bit systems usually.